kolibri/core/device/serializers.py

```diff
@@ -116,22 +116,33 @@ def create(self, validated_data):  # noqa C901
                 # We've already imported the facility to the device before provisioning
                 facility = Facility.objects.get(pk=facility_id)
                 preset = facility.dataset.preset
+                facility_created = False
             else:
                 facility = Facility.objects.create(**facility_data)
                 preset = validated_data.pop("preset")
                 facility.dataset.preset = preset
                 facility.dataset.reset_to_default_settings(preset)
+                facility_created = True
 
             custom_settings = validated_data.pop("settings")
 
-            if "on_my_own_setup" in custom_settings:
-                facility.on_my_own_setup = custom_settings.pop("on_my_own_setup")
+            if facility_created:
```

Comment:
This is great - I did wonder a bit about if we needed to run this after importing and thought (obviously incorrectly) that the values of `custom_settings` here was just populated with, if anything, the data from the imported facility so this would basically be a nothingburger